src/Events/lib.mo

```diff
@@ -81,43 +80,90 @@ module {
         #Public;
         // Denotes an event with limited access.
         #Private : Allowlist;
+        // Denotes an event with limited access to holders of a certain NFT.
+        #Holders : HolderAccess;
+    };
+
+    public type BalanceInterface = actor {
+        // Returns the amount of NFTs of the given user.
+        launchpadBalanceOf : query (user : Principal) -> async Nat;
```

Comment:
I implemented this function as a query to increase the speed. YET, this results in the limitation that the function itself can not communicate with other canisters, so the state needs to be in that canister itself.